apps/address-service/domains/common/utils/services/suggest/AbstractSuggestionProvider.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @typedef {Object} NormalizedSuggestion
+ * @property {string} country "Russia"
+ * @property {?string} region "Sverdlovsk region"
+ * @property {string} city "Yekaterinburg"
+ * @property {?string} street "Lenina"
+ * @property {string} building "66", "66a"
+ * @property {?string} unitType "room", "flat", "box"
+ * @property {?string} unitName "428", "42/8"
+ */
+
+/**
+ * @abstract
+ */
+class AbstractSuggestionProvider {
+    /**
+     * Calls to particular external service using particular client
+     * @param {string} query
+     * @param {boolean} isServerSide sometimes we need different query params for queries from front-end and backend (for dadata)
+     * @param {number} count
+     * @returns {Promise<*>}
+     * @abstract
+     * @private
+     */
+    async call ({ query, isServerSide = false, count = 20 }) {
```

Comment:
It seems like the `call` method is only for internal usage, since we have a verb `get`.
Maybe to make it private?

Reply:
I just simplified the logic. Now only the `get` method is needed.